### ISSUE TITLE
Trying to use the in-game VR keyboard to copy any text which contains the double quote character `"` or any other invalid path character does not work

### ISSUE THREAD

**Issue by Nytra** [Yellow-Dog-Man/Resonite-Issues#1206]
### Describe the bug?

Opening a keyboard in VR for a piece of text which contains the double quote character `"` and then trying to copy the text does not work. It does not copy the text and breaks the text editor until closing the keyboard.

Edit:

On looking into this further, it seems to affect text containing any invalid path characters as defined in the C# method GetInvalidPathChars() from System.IO.Path, these characters include:

`"` Double quotes
`<` Less than (or open angled bracket)
`>` Greater than (or close angled bracket)
`|` Vertical bar

### To Reproduce

1) Be in Resonite in VR
2) Click on a piece of text to edit it (In my case I used the Create New menu to make a new text)
3) Edit the text so that it includes the double quote character `"`
4) Use the keyboard 'Select All' button and then 'Copy' button to attempt to copy the text
5) Observe that the text will remain highlighted and the keyboard will no longer work (backspace or any other keys do not work), the text editor will remain broken until closing the keyboard

Alternate replication item (just text field with text): resrec:///U-Nytra/R-C216E788527851879A8402E0D93CD7B839612965CDA39CE72147ED7E96DFFFF3

### Expected behavior

The text will be copied and will not break the text editor.

### Screenshots

![20240120075338_1](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/14206961/74e4df87-eb1e-4203-be63-0d0b11a4b023)

### Resonite Version Number

2024.1.19.687

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Quest 2, Desktop

### Log Files

[DESKTOP-H976HO2 - 2024.1.19.687 - 2024-01-20 07_30_56.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13997176/DESKTOP-H976HO2.-.2024.1.19.687.-.2024-01-20.07_30_56.log)

The relevant exception seems to be:

```
Exception in RunningCoroutine: FrooxEngine.TextEditor+<EditCoroutine>d__78
System.ArgumentException: Illegal characters in path.
  at System.IO.Path.IsPathRooted (System.String path) [0x0001b] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.TextEditor+<EditCoroutine>d__78.MoveNext () [0x001f5] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.CoroutineManager.RunStep (FrooxEngine.CoroutineHandle coroutine) [0x0001d] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <d6cfefdc0ccf4b759cc2ecc37fd61287>:0 
  at FrooxEngine.DebugManager.Error (System.Object obj, System.Boolean stackTrace) [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.CoroutineManager.RunStep (FrooxEngine.CoroutineHandle coroutine) [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.CoroutineManager.ExecuteWorldQueue (System.Double deltaTime) [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.World.RefreshStep () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.World.Refresh () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <27f639639d0b4ddba65f8ad9d6d1aaf4>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <ad812d4e79624a2397d689c5d5b35796>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <ad812d4e79624a2397d689c5d5b35796>:0 
```

### Additional Context

The problem seems to be exclusive to the in-VR keyboard

### Reporters

Nytra (discord: nytra)
sctanf (discord: sctanf)

**Comment by AlanMoonbase** [comment]
Having the same issue, probably same exception.

**Comment by Frooxius** [comment]
If you think it's the same exception please provide logs. Vague information doesn't really help.

**Comment by AlanMoonbase** [comment]
That's as much info as I can give because my logs are tainted with mods. Someone else please step up.

**Comment by Frooxius** [comment]
We ask that you don't just bump issues: https://github.com/Yellow-Dog-Man/Resonite-Issues/blob/main/HOW_WE_PRIORITIZE.md

Commenting without providing more information doesn't help boost the issue. Up-voting it will, or providing more actionable information will.

**Comment by AlanMoonbase** [comment]
it's the same exception
```
9:37:55 PM.061 ( 44 FPS)	Exception in RunningCoroutine: FrooxEngine.TextEditor+<EditCoroutine>d__78
System.ArgumentException: Illegal characters in path.
  at System.IO.Path.IsPathRooted (System.String path) [0x0001b] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.TextEditor+<EditCoroutine>d__78.MoveNext () [0x001f5] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.CoroutineManager.RunStep (FrooxEngine.CoroutineHandle coroutine) [0x0001d] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <9555aed94d8841e38fa1f2f558b95641>:0 
  at FrooxEngine.DebugManager.Error (System.Object obj, System.Boolean stackTrace) [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.CoroutineManager.RunStep (FrooxEngine.CoroutineHandle coroutine) [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.CoroutineManager.ExecuteWorldQueue (System.Double deltaTime) [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.World.RefreshStep () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.World.Refresh () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <1eebf8ac462f4f8b977491e1d195d0e7>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <579215a9836c4145b540c27cc1b63d27>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <579215a9836c4145b540c27cc1b63d27>:0 
```

**Comment by ProbablePrime** [comment]
It might not actually be though.

This is why log files are important. Please reply with a log file.

**Comment by AlanMoonbase** [comment]
[ALAN-VNMBLZR - 2024.9.13.672 - 2024-09-15 21_33_46.log](https://github.com/user-attachments/files/17008854/ALAN-VNMBLZR.-.2024.9.13.672.-.2024-09-15.21_33_46.log)

**Comment by Frooxius** [comment]
I've fixed this in 2024.9.16.715, thanks!
